pkg/controller/zookeepercluster/zookeepercluster_controller.go

```diff
@@ -559,17 +592,30 @@ func (r *ReconcileZookeeperCluster) cleanUpZookeeperPVC(instance *zookeeperv1bet
 	if err != nil {
 		return err
 	}
+
 	for _, pvcItem := range pvcList.Items {
-		pvcDelete := &corev1.PersistentVolumeClaim{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      pvcItem.Name,
-				Namespace: pvcItem.Namespace,
-			},
-		}
-		err = r.client.Delete(context.TODO(), pvcDelete)
-		if err != nil {
-			return err
+		if all {
```

Comment:
A conditional switch statement can be used to clean up this nested if / else block.

Reply:
Could we instead have separate functions for clean all and clean orphan? I find the separation of functionality based on a flag error prone.

Reply:
Sure. Will do this.

Reply:
Fixed.